packages/aws-cdk-lib/aws-ec2/lib/vpc.ts

```diff
@@ -1486,6 +1594,23 @@ export class Vpc extends VpcBase {
     const natGatewayPlacement = props.natGatewaySubnets || { subnetType: SubnetType.PUBLIC };
     const natGatewayCount = determineNatGatewayCount(props.natGateways, this.subnetConfiguration, this.availabilityZones.length);
 
+    if (this.useIpv6) {
+      this.ipv6IpAddresses = props.ipv6IpAddresses ?? Ipv6IpAddresses.amazonProvided();
+
+      if (this.ipv6IpAddresses.amazonProvided) {
+        // create the IPv6 CIDR block and associate it with the VPC
+        // create a dependency of the subnets on the CIDR block
+        this.ipv6CidrBlock = new CfnVPCCidrBlock(this, 'ipv6cidr', {
+          vpcId: this.vpcId,
+          amazonProvidedIpv6CidrBlock: this.ipv6IpAddresses.amazonProvided,
+        });
+
+        this.ipv6SelectedCidr = Fn.select(0, this.resource.attrIpv6CidrBlocks);
+      }
+    }
+
+    // Add tag here too?
```

Comment:
Tags are added just above here, around line 1565: Tags.of(this).add(NAME_TAG...

I strongly suggest adding a tag, e.g. 'aws-cdk:vpc-protocol' with the selected type, similar to how subnets have a subnet type tag.